### PR TITLE
Refactor the crypto services

### PR DESCRIPTION
**Pull request by tudor-malene**
### Why this change is needed

In preparation for MainNet, we must use randomness generated inside the enclave.

### What changes were made as part of this PR

- clarify the responsibilities of the crypto services
- generate randomness for the da services
- some wiring required by the random number
- the RPC key is random. It is now exposed as an RPC endpoint which is called behind the scenes when creating an encrypted connection

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/ten-protocol/ten-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [x] PR checks reviewed and performed

### CODE REVIEW

File: go/rpc/encrypted_client.go [hunk 1]
```diff
@@ -25,11 +25,6 @@ import (
 	gethlog "github.com/ethereum/go-ethereum/log"
 )
 
-const (
-	// todo: this is a convenience for testnet testing and will eventually be retrieved from the L1
-	enclavePublicKeyHex = "034d3b7e63a8bcd532ee3d1d6ecad9d67fca7821981a044551f0f0cbec74d0bc5e"
```
Comment (BedrockSquirrel) [inline go/rpc/encrypted_client.go:30]: This key string still appears a few times in the codebase which can probably be cleaned up? In particular can probably delete the stuff about 'EnclavePublicKey' in `transaction_injector.go`

File: go/rpc/client.go [hunk 2]
```diff
@@ -17,6 +17,7 @@ const (
 
 	Health = "ten_health"
 	Config = "ten_config"
+	RPCKey = "ten_rPCKey"
```
Comment (BedrockSquirrel) [inline go/rpc/client.go:20]: Not sure if it's case sensitive anyway but this capitalisation seems ugly haha, maybe copilot did that.

Comment (tudor-malene) [reply go/rpc/client.go]: I called the method RPCKey, and the convention is `prefix_{lowerCaseFirstChar}{otherChars}`

Comment (BedrockSquirrel) [reply go/rpc/client.go]: Fair, I'd probably rename it to `RpcKey()` to fix that if it doesn't break our linter with a comment to explain it keeps the method name nice lol. I guess that method's not super public facing but a little bit.

**Review by StefanIliev545 — APPROVED** [review]
LGTM, lets open a ticket for removing the decrypt blob from any front end if there isn't one already

**Pull request merged**